jdaviz/configs/specviz/tests/test_helper.py

```diff
@@ -36,6 +37,23 @@ def test_load_spectrum1d(self):
 
         assert isinstance(data, Spectrum1D)
 
+    def test_load_hdulist(**kwargs):
+        # Create a fake fits file with a 1D spectrum for testing.
+        primary_header = fits.Header({'TELESCOP': 'Fake Telescope'})
+        primary_hdu = fits.PrimaryHDU(header=primary_header)
+
+        # Image extension HDU with a 1D spectrum
+        wavelength = np.linspace(5000, 6000, 100)
+        flux = np.random.random(100)
+        spectrum_table = fits.BinTableHDU.from_columns([
+            fits.Column(name='WAVELENGTH', array=wavelength, format='E'),
+            fits.Column(name='FLUX', array=flux, format='E')
+        ])
+        spectrum_table.header['INSTRUME'] = 'Fake Instrument'
+        fake_hdulist = fits.HDUList([primary_hdu, spectrum_table])
+
+        return fake_hdulist
```

Comment:
Rather than returning `fake_hdulist` you need to load it into specviz, using `self.spec_app.load_data` like the test below this one does with `spec_list`.